fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Transaction.java

```diff
@@ -96,6 +98,37 @@ public String computeId() {
     return Hash.hash(collectTransaction.toArray(new String[0]));
   }
 
+  /**
+   * Function that given a key pair change the sig of an input considering all the outputs
+   *
+   * @return sig other all the outputs and inputs with the public key
+   */
+  public static String computeSigOutputsPairTxOutHashAndIndex(
+      List<Output> outputs, Map<String, Integer> inputsPairs) {
+    // input #1: tx_out_hash Value //input #1: tx_out_index Value
+    // input #2: tx_out_hash Value //input #2: tx_out_index Value ...
+    // TxOut #1: LaoCoin Value​​ //TxOut #1: script.type Value //TxOut #1: script.pubkey_hash Value
+    // TxOut #2: LaoCoin Value​​ //TxOut #2: script.type Value //TxOut #2: script.pubkey_hash
+    // Value...
+    List<String> sig = new ArrayList<>();
+    Iterator<Map.Entry<String, Integer>> iteInput = inputsPairs.entrySet().iterator();
+    Iterator<Output> iteOutput = outputs.iterator();
+
+    while (iteInput.hasNext()) {
+      Map.Entry<String, Integer> current = iteInput.next();
+      sig.add(current.getKey());
+      sig.add(String.valueOf(current.getValue()));
+    }
+
+    while (iteOutput.hasNext()) {
+      Output current = iteOutput.next();
+      sig.add(String.valueOf(current.getValue()));
+      sig.add(current.getScript().getType());
+      sig.add(current.getScript().getPubkeyHash());
+    }
```

Comment:
I would propose to use the foreach expression in these scenarios.

It is basically a sugaring of what you did there.

```suggestion
    for ( Map.Entry<String, Integer> current : inputsPairs.entrySet()) {
      sig.add(current.getKey());
      sig.add(String.valueOf(current.getValue()));
    }

    for(Output current : outputs) {
      sig.add(String.valueOf(current.getValue()));
      sig.add(current.getScript().getType());
      sig.add(current.getScript().getPubkeyHash());
    }
```

It produces a code easier to read